model/modeldecoder/transaction_test.go

```diff
@@ -374,6 +375,52 @@ func TestTransactionEventDecode(t *testing.T) {
 				},
 			},
 		},
+		"with derived success outcome": {
+			input: map[string]interface{}{
+				"context": map[string]interface{}{
+					"response": map[string]interface{}{
+						"status_code": json.Number("400"),
+					},
+				},
+			},
+			e: &model.Transaction{
+				Metadata: inputMetadata,
+				ID:       id,
+				Type:     trType,
+				Name:     name,
+				TraceID:  traceID,
+				Duration: duration,
+				HTTP: &model.Http{Response: &model.Resp{
+					MinimalResp: model.MinimalResp{StatusCode: &badRequestResp},
```

Comment:
Maybe a comment here that 400 is considered success from the perspective of the server? It might be surprising otherwise, to the uninitiated.